### PR TITLE
Fix: bash variables are not quoted

### PR DESCRIPTION
**Pull request by h3ssan**
## Contribution Guidelines

* [X] I've read the [contribution guidelines](https://github.com/mailcow/mailcow-dockerized/blob/master/CONTRIBUTING.md) and wholeheartedly agree them

## What does this PR include?

This PR include correction for quoted variables in `update.sh` and `generate_config.sh` scripts as to [`SC2086: Double quote to prevent globbing and word splitting.`](https://github.com/koalaman/shellcheck/wiki/SC2086)

### Short Description

In bash, using variables as `$variable` is not correct, actually can be considered a bug, since a spaced words can consider 2 separate bash commands, such as:

```bash
VALUE="hey there"

$VALUE # here it will render `hey there` and will consider `hey` is a bash command and give it an argument which is `there`.
```

Correct form is:

```bash
VALUE="hey there"

"${VALUE}"  # here it will render `"hey there"` as one command
```

For more info about this, please refer to [`SC2086: Double quote to prevent globbing and word splitting.`](https://github.com/koalaman/shellcheck/wiki/SC2086).

###  Affected Containers

None

## Did you run tests?

Yes, I tested manually.

### What did you tested?

I have executed `update.sh` and `generate_config.sh` and they both worked, also I checked all the checks that the scripts do, all good and worked perfectly.

### What were the final results? (Awaited, got)

All good.

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -22,13 +22,13 @@ prefetch_images() {
       fi
     fi
     RET_C=0
-    until docker pull ${image}; do
+    until docker pull "${image}"; do
```
Comment (DerLinkman) [inline update.sh:25]: Does Docker handle this right?

File: update.sh [hunk 2]
```diff
@@ -22,13 +22,13 @@ prefetch_images() {
...
       RET_C=$((RET_C + 1))
       echo -e "\e[33m\nError pulling $image, retrying...\e[0m"
       [ ${RET_C} -gt 3 ] && { echo -e "\e[31m\nToo many failed retries, exiting\e[0m"; exit 1; }
       sleep 1
     done
-  done < <(git show origin/${BRANCH}:docker-compose.yml | grep "image:" | awk '{ gsub("image:","", $3); print $2 }')
+  done < <(git show "origin/${BRANCH}:docker-compose.yml" | grep "image:" | awk '{ gsub("image:","", $3); print $2 }')
```
Comment (DerLinkman) [inline update.sh:31]: Does Git handle this right?

File: update.sh [hunk 3]
```diff
@@ -109,12 +109,12 @@ migrate_docker_nat() {
...
           echo "You will need to merge your existing configuration manually or fix/delete the existing daemon.json configuration before trying the update process again."
           echo -e "Please merge the following content and restart the Docker daemon:\n"
-          echo ${NAT_CONFIG}
+          echo "${NAT_CONFIG}"
           return 1
       fi
     else
       echo "Working on IPv6 NAT, please wait..."
-      echo ${NAT_CONFIG} > /etc/docker/daemon.json
+      echo "${NAT_CONFIG}" > /etc/docker/daemon.json
```
Comment (DerLinkman) [inline update.sh:117]: Is this correctly added to the docker daemon without the quotes?

File: update.sh [hunk 4]
```diff
@@ -340,20 +340,20 @@ fi
...
-BRANCH=$(cd ${SCRIPT_DIR}; git rev-parse --abbrev-ref HEAD)
+BRANCH="$(cd "${SCRIPT_DIR}"; git rev-parse --abbrev-ref HEAD)"
 
 while (($#)); do
   case "${1}" in
     --check|-c)
       echo "Checking remote code for updates..."
-      LATEST_REV=$(git ls-remote --exit-code --refs --quiet https://github.com/mailcow/mailcow-dockerized ${BRANCH} | cut -f1)
-      if [ $? -ne 0 ]; then
+      LATEST_REV=$(git ls-remote --exit-code --refs --quiet https://github.com/mailcow/mailcow-dockerized "${BRANCH}" | cut -f1)
```
Comment (DerLinkman) [inline update.sh:349]: Git again (like above)

File: update.sh [hunk 5]
```diff
@@ -916,17 +916,17 @@ fi
 echo -e "\e[32mChecking for conflicting bridges...\e[0m"
 MAILCOW_BRIDGE=$($COMPOSE_COMMAND config | grep -i com.docker.network.bridge.name | cut -d':' -f2)
 while read NAT_ID; do
-  iptables -t nat -D POSTROUTING $NAT_ID
-done < <(iptables -L -vn -t nat --line-numbers | grep $IPV4_NETWORK | grep -E 'MASQUERADE.*all' | grep -v ${MAILCOW_BRIDGE} | cut -d' ' -f1)
+  iptables -t nat -D POSTROUTING "$NAT_ID"
+done < <(iptables -L -vn -t nat --line-numbers | grep "$IPV4_NETWORK" | grep -E 'MASQUERADE.*all' | grep -v "${MAILCOW_BRIDGE}" | cut -d' ' -f1)
```
Comment (DerLinkman) [inline update.sh:920]: Does IPTable handle this correctly?

**Review by DerLinkman — CHANGES_REQUESTED** [review]
See comments, general looks fine!

**Comment by h3ssan** [comment]
> See comments, general looks fine!

Great, I'll see them and make sure 100% again and let you know.

File: update.sh (hunk 1 above)
Comment (h3ssan) [reply update.sh:25]: Yes, docker handle this correctly.

File: update.sh (hunk 2 above)
Comment (h3ssan) [reply update.sh:31]: Yes, git handle this correctly.

File: update.sh (hunk 3 above)
Comment (h3ssan) [reply update.sh:117]: Yes, added to the docker daemon without the quotes.

File: update.sh (hunk 4 above)
Comment (h3ssan) [reply update.sh:349]: Yes, git handle this correctly.

File: update.sh (hunk 5 above)
Comment (h3ssan) [reply update.sh:920]: Yes, it does.

**Comment by h3ssan** [comment]
Since it is a bash matter, then yes. All utilities works out of the box.

**Review by DerLinkman — APPROVED** [review]
Thanks for clarification!

**Pull request merged**